### DISCUSSION TITLE
How to avoid invalidateAll() on each form action?

### DISCUSSION THREAD

**Discussion by stefandevo** [sveltejs/kit#9467] [Q&A] [answered]
It seems that after each form action, there is an invalidateAll() happening, which causes all the load function to re-run.
I have a load in the +layout.server.ts file that is executed every time I have a form that I am posting. How can this be avoided?

Can be seen here. On the First route there is a form. As long as I navigate between all forms, the load action in the layout.server file is not run. But if I post the form, the load is executed again on the layout.server.

https://stackblitz.com/edit/sveltejs-kit-template-default-gf1pq5

**Comment by teemingc** [comment, marked as answer]
We can prevent `invalidateAll` when using the `use:enhance` action on the form.
First, we pass a custom callback, and avoid calling `applyAction` when `result.type`  is a `redirect`.

```svelte
  <form method="POST" use:enhance={() => {
    return async ({ result }) => {
      if (result.type === 'redirect') {
        // redirect without invalidating
        await goto(result.location);
      } else {
        await applyAction(result);
      }
    }
  }}>
```

https://stackblitz.com/edit/sveltejs-kit-template-default-fbiwbs?file=src%2Froutes%2Ffirst%2F%2Bpage.svelte,src%2Froutes%2Ffirst%2F%2Bpage.server.ts

**Reply by stefandevo** [reply]
Thanks for the help

**Reply by stefandevo** [reply]
Just wondering if this is something that can be added in the framework, a kind of value in the enhance that you can decide to invalidate or not.

**Reply by teemingc** [reply]
Yes, you can submit an issue for it to suggest this enhancement

**Reply by NormandoHall** [reply]
Hi @s3812497 

`await applyAction(result)` not `invalidateAll`. You need to explicit run `await invalidateAll()`

[applyAction](https://kit.svelte.dev/docs/form-actions#progressive-enhancement-applyaction) says nothing about `invalidateAll()`. I was confirmed in my server.
I am wrong?

**Reply by teemingc** [reply]
@NormandoHall I think you’re right. I was confused by this at first as well. It seems that `applyAction` only invalidates if the result type is a redirect.
